### ISSUE TITLE
Question: how to get numbers in a digit form with finetuned mozilla-ai Whisper based models?

### ISSUE THREAD

**Issue by poulpor** [mozilla-ai/speech-to-text-finetune#91]
Hi, 
I apologize if it's not the right place to ask for this but I have a question regarding one of the mozilla-ai finetuned models.
 
I'm desperately searching for a french finetuned ASR/STT model to voice command Home Assistant, so I gave a try to your model [mozilla-ai/whisper-small-fr](https://huggingface.co/mozilla-ai/whisper-small-fr).

The "problem" I encounter is when I vocalize numbers, the transcribed form is words, and, apart some exceptions, Home Assistant needs digits to understand things. 
For example, I need the STT to send to HA the text "rappelle-moi de faire à manger à 11:30", but this finetuned whisper model is sending "rappelle-moi de faire à manger à onze heure trente". (in english, its "remind me to do this at 11:30).

I suppose it has something to do with Text Normalization, I surely need to add an Inverse Text Normalization (ITN) to the ASR process but how to do it? Can it be done by specific setting when finetuning?

The thing is, Whisper/Faster Whisper tiny/base/small are al set in a manner that I directly get digit numbers, but every finetuned french variations I tried treat numbers as words instead of digits like the original model..?

Thank you for any help you'll be able to provide.

**Comment by Kostis-S-Z** [comment]
Hey @poulpor , 

Thank you for your question, and so sorry for the long delay to reply!

My colleague, Kathy Reid (@moz-kathyreid) , provided an answer that might be of help to you:
> The base Whisper models apply a post-processing normalization step that converts written-out numbers to digits, but fine-tuned models often lose this behavior because the fine-tuning data reinforces the word form. 
The simplest approach is to apply ITN as a post-processing step after transcription, completely separate from the model itself. The `nemo_text_processing` library from NVIDIA NeMo has solid French ITN support.

Let me know if that works or you have any other questions!

**Comment by poulpor** [comment]
Hi, 
as the old saying says, better late than never, so thank you very much for your kind reply :)

I have to admit, I tried the NeMo way but lack the knowledge to add this to my Whisper/Home Assistant dockerized environment. 
Hopefully, soon after I asked this question, I stumbled upon the one and only french fine-tuned small whisper model able to keep this base Whisper behavior, if anyone is interested in the future, it's named [keypa/whisper-3-mls-fr](https://huggingface.co/keypa/whisper-3-mls-fr). 
There's also a more recent [hip94/52Hz-small-fr-v2](https://huggingface.co/hip94/52Hz-small-fr-v2), so french people can now command their Home Assistant instance in native language and numbers to digits!

**Comment by Kostis-S-Z** [comment]
Great to hear that you found existing models that already support this! 

I will close the issue for now, but feel free to ask more questions or for more help if needed. 

Also note that we have extended some of our work in[ this repository](https://github.com/Mozilla-Data-Collective/speech-to-text-finetune) to also include finetuning of the MMS model.

For more ASR related questions you are also welcome to ask at Mozilla's [Common Voice](https://commonvoice.mozilla.org/en) community on [Discord](https://discord.gg/4TjgEdq25Y), [Matrix](https://chat.mozilla.org/#/room/#common-voice:mozilla.org) or [Discourse](https://discourse.mozilla.org/c/voice/239).
